src/alph_puzzle.c

```diff
@@ -0,0 +1,971 @@
+
+    BG_ClearCharDataRange(4, 32, 0, data->heapId);
+    BG_ClearCharDataRange(7, 32, 0, data->heapId);
+    BG_ClearCharDataRange(0, 32, 0, data->heapId);
+    BG_ClearCharDataRange(3, 64, 0, data->heapId);
+}
+
+static void ov110_021E6348(AlphPuzzleData *data) {
+    FreeBgTilemapBuffer(data->bgConfig, 3);
+    FreeBgTilemapBuffer(data->bgConfig, 2);
+    FreeBgTilemapBuffer(data->bgConfig, 0);
+    FreeBgTilemapBuffer(data->bgConfig, 7);
+    FreeBgTilemapBuffer(data->bgConfig, 6);
+    FreeBgTilemapBuffer(data->bgConfig, 4);
+    FreeToHeap(data->bgConfig);
+    
+    u16 *unkPtr = (u16 *)(0x04000304);
+    *unkPtr |= (1 << 15);
+}
+
+static void ov110_021E6394(AlphPuzzleData *data) {
+    NARC *narc = NARC_New(NARC_a_1_7_2, data->heapId);
+    data->palette = PaletteData_Init(data->heapId);
+    
+    PaletteData_AllocBuffers(data->palette, 0, (1 << 8), data->heapId);
+    PaletteData_AllocBuffers(data->palette, 1, 256, data->heapId);
+    PaletteData_AllocBuffers(data->palette, 2, 256, data->heapId);
+    
+    sub_02003220(data->palette, narc, 10, data->heapId, 0, (1 << 8), 0, 0);
+    sub_02003220(data->palette, narc, 10, data->heapId, 1, 256, 0, 0);
+    sub_02003220(data->palette, narc, 0, data->heapId, 2, 256, 0, 0);
+    
+    GfGfxLoader_LoadCharDataFromOpenNarc(narc, 11, data->bgConfig, GF_BG_LYR_SUB_3, 0, 0, 0, data->heapId);
+    GfGfxLoader_LoadScrnDataFromOpenNarc(narc, 14, data->bgConfig, GF_BG_LYR_SUB_3, 0, 0, 0, data->heapId);
+    GfGfxLoader_LoadScrnDataFromOpenNarc(narc, 15, data->bgConfig, GF_BG_LYR_SUB_2, 0, 0, 0, data->heapId);
+    GfGfxLoader_LoadCharDataFromOpenNarc(narc, 11, data->bgConfig, GF_BG_LYR_MAIN_3, 0, 0, 0, data->heapId);
+    
+    GfGfxLoader_LoadScrnDataFromOpenNarc(narc, 12, data->bgConfig, GF_BG_LYR_MAIN_3, 0, 0, 0, data->heapId);
+    data->unkD8 = GfGfxLoader_GetScrnDataFromOpenNarc(narc, 13, 0, &data->screenData, data->heapId);
+    
+    NARC_Delete(narc);
+    
+    PaletteData_LoadNarc(data->palette, NARC_a_0_3_8, data->frame + 26, data->heapId, 0, 32, 80);
+    PaletteData_LoadNarc(data->palette, NARC_graphic_font, 8, data->heapId, 0, 32, 64);
+    
+    LoadUserFrameGfx2(data->bgConfig, GF_BG_LYR_MAIN_0, 1, 5, data->frame, data->heapId);
+    sub_02003B50(data->palette, 1);
+    sub_0200398C(data->palette);
+}
+
+static void ov110_021E6544(AlphPuzzleData *data) {
+    FreeToHeap(data->unkD8);
+    PaletteData_FreeBuffers(data->palette, 2);
+    PaletteData_FreeBuffers(data->palette, 1);
+    PaletteData_FreeBuffers(data->palette, 0);
+    PaletteData_Free(data->palette);
+}
+
+static void ov110_021E6580(AlphPuzzleData *data) {
+    FontID_Alloc(4, data->heapId);
+    
+    data->msgData = NewMsgDataFromNarc(MSGDATA_LOAD_DIRECT, NARC_msgdata_msg, 2, data->heapId);
+    data->messageFormat = MessageFormat_New_Custom(6, 16, data->heapId);
+    data->unk30 = String_New(0x80, data->heapId);
+    
+    data->unk34 = NewString_ReadMsgData(data->msgData, 0);
```

Comment:
(include msgdata/msg/msg_0002.h)
```suggestion
    data->unk34 = NewString_ReadMsgData(data->msgData, msg_0002_00000);
```